cmd/utils/dev.go

```diff
@@ -387,6 +387,11 @@ func GetApp(ctx context.Context, dev *model.Dev, c kubernetes.Interface, isRetry
 	app, err := apps.Get(ctx, dev, dev.Namespace, c)
 	if err != nil {
 		if !oktetoErrors.IsNotFound(err) {
+			if len(dev.Selector) > 0 {
```

Comment:
the check below would never be true because it would always end up here, I figure we can merge them and also avoid the double check on "not found"

Reply:
Why we need to include this check here? This if is whenever we get an error which is a not found. I mean, if the error is a transient error connecting with K8s, it might exist an application that matches with the labels but we weren't able to retrieve it for other reasons.

The other cases only happen when we got a not-found error

Reply:
What I've observed is that because there is this initial check for NotFound errors, [my change](https://github.com/okteto/okteto/pull/3436/files#diff-70f6fa6cd3171b4938b73394bb9432e12df9ae8da9203c639799aad23a926161R136) would always end up in this conditional branch instead of the one below with the Selector.

My understandig is that these 2 check against the same thing:

```go
if !oktetoErrors.IsNotFound(err) {
```

and 

```go
if err == oktetoErrors.ErrNotFound {
```

so they could be merged together. Maybe I'm misisng something I can't see yet. 🤔

Reply:
But in line `389` we are checking that the error is **not** a `NotFound`. So it might be that there is a dev with the specified label selector, but it could not be retrieved for whatever reason.

Reply:
good catch @ifbyol, I have missed the `!` 😄 Fixing now.

Reply:
I've reverted this change. I was too excited to open the PR 😅

Reply:
I'm moving the PR to draft to take some time to review properly because I think it's not printing the right error message.

Reply:
All good, I was slightly confused because in the issue the expected error message was different from what I think should be displayed.